indra/llcommon/llsd.cpp

```diff
@@ -872,6 +916,155 @@ const LLSD::Binary& LLSD::asBinary() const  { return safe(impl).asBinary(); }
 
 const LLSD::String& LLSD::asStringRef() const { return safe(impl).asStringRef(); }
 
+LLSD::String LLSD::asXMLRPCValue() const { return "<value>" + safe(impl).asXMLRPCValue() + "</value>"; }
```

Comment:
This is a wrong place for this logic, I suggest following LlsdFromJson and LlsdToJson example and move this to a separate llsdxmlrpc file or follow LLSDXMLParser's example and do an LLSDXMLRPCParser.

Reply:
I would like to re-use the existing type hierarchy implemented in LLSD instead of creating the mirror outside

Reply:
But it isn't a type, it's xml parsing. 

P.S. I think llsd files are shared between viewer and server so it might be better to minimize changes.